Astro 4 support

This integration is not compatible with Astro 4 that was released today. When installing/updating the package the following error is shown:

```
npm install @sanity/astro
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While hidden@0.0.1
npm ERR! Found: astro@4.0.1
npm ERR! node_modules/astro
npm ERR!   astro@"^4.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer astro@"^2.0.0 || ^3.0.0" from @sanity/astro@2.1.4
npm ERR! node_modules/@sanity/astro
npm ERR!   @sanity/astro@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

It looks that this package is locked to Astro v2 and v3. I think Astro 4 also has some breaking changes for plugin authors. For example they are now using Vite 5.

Their upgrade guide:
https://docs.astro.build/en/guides/upgrade-to/v4/

--- a/packages/sanity-astro/package.json
+++ b/packages/sanity-astro/package.json
@@ -49,7 +49,7 @@
   },
   "peerDependencies": {
     "@sanity/client": "^6.4.12",
-    "astro": "^2.0.0 || ^3.0.0",
+    "astro": "^2.0.0 || ^3.0.0 || ^4.0.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "sanity": "*"

--- a/packages/sanity-astro/src/index.ts
+++ b/packages/sanity-astro/src/index.ts
@@ -39,7 +39,8 @@ export function sanityIntegration(
         // only load this route if `studioBasePath` is set
         if (resolvedOptions.studioBasePath) {
           injectRoute({
-            entryPoint: "@sanity/astro/studio/studio-route.astro",
+            entryPoint: "@sanity/astro/studio/studio-route.astro", // Astro <= 3
+            entrypoint: "@sanity/astro/studio/studio-route.astro", // Astro > 3
             pattern: `/${resolvedOptions.studioBasePath}/[...params]`,
             prerender: false,
           });

